webpackjs/components/DocklessData/index.js

```diff
@@ -146,37 +177,62 @@ class DocklessData extends Component {
   }
 
   getDeviceValue(mode, month, year) {
-    const { deviceCountData } = this.state;
+    const { deviceCountData, allTimeDeviceCountData } = this.state;
+    const monthYear = `${month}_${year}`;
 
     // return 0 when the API hasn't responded yet but the HTML needs to render
-    if (!deviceCountData) {
+    if (!deviceCountData || !allTimeDeviceCountData) {
       return 0;
     }
 
     const scooterDataByMonth = deviceCountData.scooter[`${month}_${year}`];
     const bicycleDataByMonth = deviceCountData.bicycle[`${month}_${year}`];
+    const scooterDataAllTime = allTimeDeviceCountData.scooter;
+    const bicycleDataAllTime = allTimeDeviceCountData.bicycle;
 
     if (mode === "all") {
-      // if both modes are undefined, don't try to sum them to return NaN.
-      if (
-        typeof scooterDataByMonth === "undefined" &&
-        typeof bicycleDataByMonth === "undefined"
-      ) {
-        return "no data";
-      }
-
+      // if both modes are undefined, don't try to sum them to return NaN,
       // else sum the modes
-      return (scooterDataByMonth || 0) + (bicycleDataByMonth || 0);
+      if (monthYear === "ALL_TIME") {
+        if (
+          typeof scooterDataAllTime === "undefined" &&
+          typeof bicycleDataAllTime === "undefined"
+        ) {
+          return "no data";
+        } else {
+        return ((scooterDataAllTime || 0 ) + (bicycleDataAllTime || 0));
+        }
+      } else {
+        if (
+          typeof scooterDataByMonth === "undefined" &&
+          typeof bicycleDataByMonth === "undefined"
```

Comment:
Similar idea here. There's lots of if/else if nesting. And I'm having trouble breaking down the logic of whats supposed to happen in each case. Is there way we could turn some of these conditions into understandable variables that are booleans? Maybe that could help break up the logic too? And maybe thinking about organizing the blocks by similarities in what they return.

```js
// set the more verbose type check code to a boolean variable for easier reading
let isAllTimeModeDataUndefined = typeof scooterDataAllTime === "undefined" && typeof bicycleDataAllTime === "undefined"
let isModeDataByMonthUndefined = typeof scooterDataByMonth === "undefined" && typeof bicycleDataByMonth === "undefined"

// then break down the if/else blocks to be smaller
if (isAllTimeModeDataUndefined || isModeDataByMonthUndefined) {
  return "no data";
} else if (monthYear === "ALL_TIME") {
  return ((scooterDataAllTime || 0 ) + (bicycleDataAllTime || 0));
} else {
  return (scooterDataByMonth || 0) + (bicycleDataByMonth || 0);
}

```

This one is tricky (and the second half in the else below too) so happy to talk through it and pair together to find a good solution.